src/CodeGen_LLVM.cpp

```diff
@@ -1327,22 +1312,21 @@ void CodeGen_LLVM::visit(const Div *op) {
             multiplier = IntegerDivision::table_s8[*const_int_divisor][2];
             shift      = IntegerDivision::table_s8[*const_int_divisor][3];
         }
-
-        Value *val = codegen(op->a);
-
+        Expr num = op->a;
         // Make an all-ones mask if the numerator is negative
-        Value *sign = builder->CreateAShr(val, codegen(make_const(op->type, op->type.bits()-1)));
-        // Flip the numerator bits if the mask is high.
-        Value *flipped = builder->CreateXor(sign, val);
-
-        // Grab the multiplier.
-        Value *mult = ConstantInt::get(llvm_type_of(op->type), multiplier);
-
-        // Widening multiply, keep high half, shift
-        val = unsigned_mulhi_shr(flipped, mult, shift);
-
-        // Maybe flip the bits again
-        value = builder->CreateXor(val, sign);
+        Expr sign = num >> make_const(op->type, op->type.bits() - 1);
+        // Take the absolute value.
```

Comment:
This is not taking the absolute value. The operation it's doing is select(x < 0, -x-1, x)